packages/openai-compatible/src/openai-compatible-chat-language-model.test.ts

```diff
@@ -646,32 +646,57 @@ describe('doGenerate', () => {
       expect(warnings).toEqual([]);
     });
 
-    it('should respect the includeUsage option', async () => {
+    it('should respect the reasoningEffort setting', async () => {
       prepareJsonResponse({ content: '{"value":"test"}' });
 
       const model = new OpenAICompatibleChatLanguageModel(
         'gpt-4o-2024-08-06',
-        {},
+        { reasoningEffort: 'high' },
         {
           provider: 'test-provider',
           url: () => 'https://my.api.com/v1/chat/completions',
           headers: () => ({}),
-          includeUsage: true,
         },
       );
 
-      const { warnings } = await model.doStream({
+      await model.doGenerate({
         inputFormat: 'prompt',
         mode: { type: 'regular' },
         prompt: TEST_PROMPT,
       });
 
       const body = await server.calls[0].requestBody;
 
-      expect(body.stream).toBe(true);
-      expect(body.stream_options).toStrictEqual({ include_usage: true });
+      expect(body.reasoning_effort).toBe('high');
+    });
 
-      expect(warnings).toEqual([]);
+    it('should respect the reasoningEffort provider option', async () => {
```

Comment:
I could try to add an xAI specific test, but it would probably end up being the same as this one